logbus/solvermon/vertexmon.go

```diff
@@ -37,111 +37,147 @@ type vertexMonitor struct {
 	isCanceled     bool
 }
 
-var reErrExitCode = regexp.MustCompile(`^(?:process ".*" did not complete successfully|error calling LocalhostExec): exit code: (?P<exit_code>[0-9]+)$`)
-var reErrNotFound = regexp.MustCompile(`^failed to calculate checksum of ref ([^ ]*): (.*)$`)
-var reHint = regexp.MustCompile(`^(?P<msg>.+?):Hint: .+`)
+var reErrExitCode = regexp.MustCompile(`(?:process ".*" did not complete successfully|error calling LocalhostExec): exit code: (?P<exit_code>[0-9]+)$`)
 
-func (vm *vertexMonitor) Write(dt []byte, ts time.Time, stream int) (int, error) {
-	if stream == BuildkitStatsStream {
-		stats, err := vm.ssp.Parse(dt)
+func getExitCode(errString string) uint64 {
+	if reErrExitCode.MatchString(errString) {
+		matches, _ := stringutil.NamedGroupMatches(errString, reErrExitCode)
+		exitCodeMatch := matches["exit_code"][0]
+		exitCode, err := strconv.ParseUint(exitCodeMatch, 10, 32)
 		if err != nil {
-			return 0, errors.Wrap(err, "failed decoding stats stream")
+			return 0
```

Comment:
this feels wrong: why are we returning an error code of `0`, which indicates things are ok when in fact they aren't.

I would change this function to return (uint64, error), or maybe even simply (int, error) unless there's a specific reason you need 64 bits or unsigned.

Reply:
fixed and moved to int